SonarScanner.Shim/PropertiesFileGenerator.cs

```diff
@@ -220,6 +247,22 @@ private static ProjectInfoAnalysisResult ProcessProjectInfoFiles(IEnumerable<Pro
                     {
                         string fxCopReport = TryGetFxCopReport(projectInfo, logger);
                         string vsCoverageReport = TryGetCodeCoverageReport(projectInfo, logger);
+                        if (projectInfo.Encoding != null)
```

Comment:
I guess you can keep the code like this. We will need to do some refactoring at some point because some things can really be improved.